urle27f76cced_iconmaster5326_HPPPlus_tgz-pJ8-com_iconmaster_hppplus_HPPPlusJ8/src/com/iconmaster/hppplus/parse/element/block/ElementFor.java

```diff
@@ -0,0 +1,177 @@
+
+package com.iconmaster.hppplus.parse.element.block;
+
+import com.iconmaster.hppplus.HPPPlus;
+import com.iconmaster.hppplus.SourceRange;
+import com.iconmaster.hppplus.exception.HPPPlusException;
+import com.iconmaster.hppplus.exception.parse.DoExpectedException;
+import com.iconmaster.hppplus.exception.parse.EndExpectedException;
+import com.iconmaster.hppplus.exception.parse.IllegalArgumentCountException;
+import com.iconmaster.hppplus.parse.Element;
+import com.iconmaster.hppplus.parse.Parser;
+import com.iconmaster.hppplus.tokenize.Token;
+import com.iconmaster.hppplus.tokenize.element.TokenWord;
+import java.util.ArrayList;
+
+/**
+ *
+ * @author iconmaster
+ */
+public class ElementFor implements Element {
+    public class ForData {
+        public Element var;
+        public ArrayList<Element> block;
+        public Element goal;
+        public Element step;
+        public ForData(Element var,Element goal,Element step,ArrayList<Element> block) {
+            this.var = var;
+            this.goal = goal;
+            this.step = step;
+            this.block = block;
+        }
+    }
+        
+    private SourceRange range;
+    private Element var;
+    private ArrayList<Element> block;
+    private int blockSize;
+    private Element goal;
+    private Element step;
+    
+    public ElementFor() {
+        
+    }
+    
+    public ElementFor(Element var,Element goal,Element step,ArrayList<Element> block,int blockSize,SourceRange range) {
+        this.range = range;
+        this.var = var;
+        this.goal = goal;
+        this.step = step;
+        this.block = block;
+        this.blockSize = blockSize;
+    }
+
+    @Override
+    public boolean isMatch(ArrayList parseList, int at) {
+        if (at+3>=parseList.size()) {return false;}
+        if (parseList.get(at) instanceof TokenWord) {
+            Token e = (Token) parseList.get(at);
+            return "FOR".equalsIgnoreCase((String) e.getParsedContent());
+        }
+        return false;
+    }
+
+    @Override
+    public Element match(ArrayList parseList, int at) throws HPPPlusException {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 25. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 114-145 into a separate method.